src/box/txn.c

```diff
@@ -407,16 +433,8 @@ txn_rollback_to_svp(struct txn *txn, struct stailq_entry *svp)
 	stailq_reverse(&rollback);
 	stailq_foreach_entry(stmt, &rollback, next) {
 		txn_rollback_one_stmt(txn, stmt);
-		if (stmt->row != NULL && stmt->row->replica_id == 0) {
-			assert(txn->n_new_rows > 0);
-			txn->n_new_rows--;
-			if (stmt->row->group_id == GROUP_LOCAL)
-				txn->n_local_rows--;
-		}
-		if (stmt->row != NULL && stmt->row->replica_id != 0) {
-			assert(txn->n_applier_rows > 0);
-			txn->n_applier_rows--;
-		}
+		if (!stmt->is_temporary && stmt->row != NULL)
+			txn_update_row_counts(txn, stmt, TXN_ROW_COUNT_DEC);
```

Comment:
Just inline `+1` or `-1`. Enum is an overkill for this. Also introduction of `txn_update_row_counts()` is another good candidate for a preparatory commit.

Reply:
done